nerfstudio/models/splatfacto.py

```diff
@@ -660,17 +707,25 @@ def get_outputs(self, camera: Cameras) -> Dict[str, Union[torch.Tensor, List]]:
             return {}
         assert camera.shape[0] == 1, "Only one camera at a time"
 
+        out = {}
+
         # get the background color
+        dims = self.config.nd_dim if self.config.use_nd else 3
         if self.training:
             if self.config.background_color == "random":
                 background = torch.rand(3, device=self.device)
+                nd_background = torch.rand(dims, device=self.device)
```

Comment:
Using random background for training only really makes sense for RGB since the gradients help opacity inflate into uncovered regions of the scene. I think the behavior for feature vectors should be to always use a torch.zeros background, then normalize with alpha similar to depth. (or maybe this step can be left out as optional if people want to add it)